src/main/java/CommandHandler.java

```diff
@@ -0,0 +1,135 @@
+        String[] inputArguments = input.split(" ", 2);
+        String commandString = inputArguments[0];
+        String argumentString = inputArguments.length > 1 ? inputArguments[1] : "";
+
+        Command command = Command.parseString(commandString);
+
+        processCommand(command, argumentString);
+    }
+
+    private void processCommand(Command command, String argumentString) {
+        if (command == null) {
+            handleInvalidCommand();
+            return;
+        }
+
+        printLine();
+
+        switch (command) {
+            case LIST:
+                handleListCommand();
+                break;
+            case MARK:
+                handleMarkCommand(argumentString);
+                break;
+            case UNMARK:
+                handleUnmarkCommand(argumentString);
+                break;
+            case TODO:
+                handleTodoCommand(argumentString);
+                break;
+            case DEADLINE:
+                handleDeadlineCommand(argumentString);
+                break;
+            case EVENT:
+                handleEventCommand(argumentString);
+                break;
+        }
+
+        printLine();
+    }
+
+    private void printLine(){
+        System.out.println("-".repeat(LINE_LENGTH));
+    }
+
+    private void handleListCommand() {
+        taskManager.listTasks();
+    }
+
+    private int parseTaskIndex(String taskIndex) {
+        try {
+            return Integer.parseInt(taskIndex.trim()) - 1;
+        } catch (Error e) {
+            return -1;
+        }
+    }
+
+    private void handleMarkCommand(String taskIndex) {
+        int markIndex = parseTaskIndex(taskIndex);
+
+        if (markIndex == -1) {
+            System.out.println("Invalid mark command. Please provide a valid task number.");
+            return;
+        }
+
+        taskManager.markTask(markIndex);
+    }
+
+    private void handleUnmarkCommand(String taskIndex) {
+        int unmarkIndex = parseTaskIndex(taskIndex);
+
+        if (unmarkIndex == -1) {
+            System.out.println("Invalid unmark command. Please provide a valid task number.");
+            return;
+        }
+
+        taskManager.unmarkTask(unmarkIndex);
+    }
+
+    private void handleTodoCommand(String description) {
+        if (description.isEmpty()){
+            System.out.println("Invalid todo command. Please provide a task description.");
+            return;
+        }
+
+        taskManager.addTodo(description);
+    }
+
+    private String[] parseArguments(String argumentString, String... delimiters) {
+        String splitBy = String.join("|", delimiters);
+        return argumentString.split(splitBy);
+    }
+
+    private void handleDeadlineCommand(String argumentString) {
+        String[] arguments = parseArguments(argumentString, " /by ");
+
+        if (arguments.length != 2) {
+            System.out.println("Invalid deadline command. Please provide a task description and a deadline using '/by'.");
```

Comment:
Should you extract out the error messages for lines 91, 107 and 121? All these seem to be doing a similar thing of handling invalid inputs.